pkg/migration-tool/writer/writer.go

```diff
@@ -74,12 +88,24 @@ func (rw *RemoteWrite) Run(errChan chan<- error) {
 				if !ok {
 					return
 				}
-				blockRef.SetDescription("preparing to push", 1)
-				ts = timeseriesRefToTimeseries(blockRef.MergeProgressSeries(rw.progressTimeSeries))
-				buf = []byte{}
-				if err = rw.sendSamplesWithBackoff(context.Background(), ts, &buf); err != nil {
-					errChan <- fmt.Errorf("remote-write run: %w", err)
-					return
+				blockRef.UpdatePBarMax(blockRef.PBarMax() + rw.shardsSet.num)
+				{
+					// Pushing data to remote-write storage.
+					blockRef.SetDescription("preparing to push", 1)
+					shards.scheduleTS(timeseriesRefToTimeseries(blockRef.Series()))
+					blockRef.SetDescription("pushing ...", 1)
+					var pushed int
+					for i := 0; i < shards.num; i++ {
+						pushed = waitForSig(blockRef, pushed, shards.num)
+					}
+				}
+				{
```

Comment:
why these extra paranthesis?

Reply:
It is just for easier reading. Each block represents a (pushed + waitForSig). So, in order to make it easier to understand the two push operations and differentiate them, I kept them as different blocks (of same nature).

Do you think this is not that beneficial?

Reply:
Personally, I don't like this style. Any time a block like this comes up, it usually implies it's better as a factored-out function and/or without the block. But I'll leave this up to you.